mobile/src/main/java/rs/readahead/washington/mobile/views/collect/widgets/WidgetFactory.java

```diff
@@ -46,6 +49,13 @@ public static QuestionWidget createWidgetFromPrompt(FormEntryPrompt fep, Context
         appearance = appearance.toLowerCase(Locale.ENGLISH);
 
         QuestionWidget questionWidget;
+        if (isTellaLocationHiddenField(context, fep)) {
```

Comment:
I think this two new conditions (checking for both hidden field) can be moved to begging of the method - not for speed but readability.